### PR TITLE
[MI-2749]:Fixed issue #9 of link filter plugin "Fix Lint errors"

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
**Summary** 

- Updated the golangci.yml file according to the GitHub plugin and fixed the lint errors as well.

**Issue** 

- [Issue #9](https://github.com/mattermost/mattermost-plugin-link-filter/issues/9)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -10,8 +10,6 @@ linters-settings:
     simplify: true
   goimports:
     local-prefixes: github.com/Brightscout/mattermost-plugin-link-filter
-  golint:
-    min-confidence: 0
   govet:
     check-shadowing: true
     enable-all: true
```
Comment (hanzei) [inline .golangci.yml:15]: There is some config missing: https://github.com/mattermost/mattermost-plugin-demo/blob/1c5c5063e69ba4a86c586fa2641a24c10dbc2360/.golangci.yml#L10-L14

File: server/configuration.go [hunk 2]
```diff
@@ -23,11 +23,11 @@ import (
 // If you add non-reference types to your configuration struct, be sure to rewrite Clone as a deep
 // copy appropriate for your types.
 type configuration struct {
-	RejectPlainLinks             bool
 	AllowedProtocolListLink      string
 	AllowedProtocolListPlainText string
 	CreatePostWarningMessage     string
 	EditPostWarningMessage       string
+	RejectPlainLinks             bool
```
Comment (hanzei) [inline server/configuration.go]: Unrelated change?

Comment (Kshitij-Katiyar) [reply server/configuration.go]: @hanzei This is to fix the lint error ```fieldalignment: struct with 40 pointer bytes could be 8```.

Comment (hanzei) [reply server/configuration.go]: Does that error still get thrown if `fieldalignment` is disabled via `.golangci.yml`

Comment (Kshitij-Katiyar) [reply server/configuration.go]: @hanzei Reverted these changes back.

**Review by hanzei — APPROVED** [review]
